src/foam/u2/AddressDetailView.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'AddressDetailView',
+  extends: 'foam.u2.View',
+
+  documentation: 'Address Detail View',
+
+  requires: [ 'foam.nanos.auth.Address' ],
+
+  css: `
+    ^ {
+      height: auto;
+    }
+    ^ .property-type {
+      width: 120px;
+      margin-left: 180px;
+      margin-bottom: 10px;
+    }
+    ^ .property-verified {
+      margin-left: 20px;
+      margin-bottom: 10px;
+    }
+    ^ .property-structure {
+      margin-left: 215px;
+      margin-bottom: 10px;
+    }
+    ^ .property-streetNumber {
+      width: 110px;
+      margin-bottom: 10px;
+    }
+    ^ .property-streetName {
+      width: 160px;
+      margin-left: 114px;
+      margin-bottom: 10px;
+    }
+    ^ .property-address {
+      width: 200px;
+      margin-bottom: 10px;
+    }
+    ^ .property-address2 {
+      width: 200px;
+      margin-left: 23px;
+      margin-bottom: 10px;
+    }
+    ^ .property-suite {
+      width: 90px;
+      margin-left: 60px;
+      margin-bottom: 10px;
+    }
+    ^ .property-city {
+      width: 110px;
+      margin-bottom: 10px;
```

Comment:
I see a lot of repeated styles here, maybe you can combine a few of them together. A good tip as well is to use the html element as a css identifier, for example to place a style on all .start('p') you can do ^ p{ width: 110px }. This will provide all 'p's the width of 110px.

Reply:
You can also set the "displayWidth: numberOfChars" property of the Properties in Address and then let the fields size themselves to the proper size. Involves less code, no CSS, and is transferable to Java and Swift clients.